src/test/skript/tests/general/EquippableComponents.sk

```diff
@@ -0,0 +1,168 @@
+	make {_item} dispensable
+	assert {_item} is dispensable with "ItemStack should be dispensable"
+	make {_item} undispensable
+	assert {_item} is not dispensable with "ItemStack should not be dispensable"
+	make {_item} lose durability when hurt
+	assert {_item} will lose durability when hurt with "ItemStack should lose durability when hurt"
+	make {_item} not lose durability when hurt
+	assert {_item} will not lose durability when hurt with "ItemStack should not lose durability when hurt"
+
+	parse if method {@method} exists:
+		make {_item} equip on interaction
+		assert {_item} can equip on interaction with "ItemStack should equip on interaction"
+		make {_item} not equip on interaction
+		assert {_item} can not equip on interaction with "ItemStack should not equip on interaction"
+
+test "equippable components - copy" when running minecraft "1.21.2":
+	set {_item} to a golden helmet
+	set {_component} to the copied equippable component of {_item}
+	set the equip sound of {_component} to {@sound}
+	assert the equip sound of {_component} is {@sound} with "Equip sound of copy was not set"
+	set the equipment slot of {_component} to {@slot}
+	assert the equipment slot of {_component} is {@slot} with "Equipment slot of copy was not set"
+	set the model key of {_component} to {@model}
+	assert the model key of {_component} is {@model} with "Model of copy was not set"
+	set the allowed entities of {_component} to {@allowed}
+	assert the allowed entities of {_component} is {@allowed} with "Allowed entities of copy was not set"
+	set the camera overlay of {_component} to {@camera}
+	assert the camera overlay of {_component} is {@camera} with "Camera overlay of copy was not set"
+	make {_component} swappable
+	assert {_component} is swappable with "Copy should be swappable"
+	make {_component} unswappable
+	assert {_component} is not swappable with "Copy should not be swappable"
+	make {_component} dispensable
+	assert {_component} is dispensable with "Copy should be dispensable"
+	make {_component} undispensable
+	assert {_component} is not dispensable with "Copy should not be dispensable"
+	make {_component} lose durability when hurt
+	assert {_component} will lose durability when hurt with "Copy should lose durability when hurt"
+	make {_component} not lose durability when hurt
+	assert {_component} will not lose durability when hurt with "Copy should not lose durability when hurt"
+
+	parse if method {@method} exists:
+		make {_component} equip on interaction
+		assert {_component} can equip on interaction with "Copy should equip on interaction"
+		make {_component} not equip on interaction
+		assert {_component} can not equip on interaction with "Copy should not equip on interaction"
+
+test "equippable components - slot" when running minecraft "1.21.2":
+	set {_gui} to a chest inventory with 1 row
+	set slot 1 of {_gui} to a diamond helmet
+	set {_item} to slot 1 of {_gui}
```

Comment:
this never actually tests slots. This line evaluates the slot and the var is just an item.
You'll need to consistently use `slot 1 of {_gui}` to properly test.